msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/MzMLSpectrum.java

```diff
@@ -32,11 +32,11 @@
 
 public class MzMLSpectrum implements MsScan {
 	private HashMap<String, String> cvParamValues;
-	private ArrayList<Long> binaryDataPositions;
+	private ArrayList<MzMLBinaryDataInfo> binaryDataInfoArray;
```

Comment:
I think you don't need ArrayLists here - each scan has m/z array and intensity array, and that's it. There will always be two arrays.